src/state_machine.zig

```diff
@@ -448,10 +452,6 @@ pub fn StateMachineType(comptime Storage: type) type {
                     chain_broken = false;
                 }
             }
-            // TODO client.zig: Validate that batch chains are always well-formed and closed.
-            // This is programming error and we should raise an exception for this in the client ASAP.
-            assert(chain == null);
-            assert(chain_broken == false);
```

Comment:
We should leave these in, since now they should never be hit, unless we have a bug in our linked event code above, or unless we're not cleaning up for the `linked_event_chain_open` result.